bin/tx-prover/src/commands/mod.rs

```diff
@@ -0,0 +1,108 @@
+use clap::Parser;
+use init::Init;
+use proxy::StartProxy;
+use serde::{Deserialize, Serialize};
+use worker::StartWorker;
+
+pub mod init;
+pub mod proxy;
+pub mod worker;
+
+/// Configuration of the proxy.
+///
+/// It is stored in a TOML file, which will be created by the `init` command.
+/// It allows manual modification of the configuration file.
+#[derive(Serialize, Deserialize)]
+pub struct ProxyConfig {
+    /// List of workers to start
+    pub workers: Vec<WorkerConfig>,
+    /// Host of the proxy.
+    pub host: String,
+    /// Port of the proxy.
+    pub port: u16,
+    /// Timeout in seconds.
+    pub timeout_secs: u64,
+    /// Connection timeout in seconds.
+    pub connection_timeout_secs: u64,
+    /// Maximum number of items in the queue.
+    pub max_queue_items: usize,
```

Comment:
I would expand the comments here a bit. For example, does `timeout_secs` means overall request timeout or timeout for a request to a particular worker.